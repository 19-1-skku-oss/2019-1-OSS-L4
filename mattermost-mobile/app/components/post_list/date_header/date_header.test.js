// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

/* eslint-disable max-nested-callbacks */

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import DateHeader from './date_header.js';

describe('DateHeader', () => {
    const baseProps = {
        theme: Preferences.THEMES.default,
        timeZone: null,
    };

    describe('component should match snapshot', () => {
        it('without suffix', () => {
            const props = {
                ...baseProps,
                date: 1531152392,
            };
            const wrapper = shallow(
                <DateHeader {...props}/>,
                {context: {intl: {formatMessage: jest.fn()}}},
            );

            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('with suffix', () => {
            const props = {
                ...baseProps,
                date: 1531152392,
            };
            const wrapper = shallow(
                <DateHeader {...props}/>,
                {context: {intl: {formatMessage: jest.fn()}}},
            );

            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('when timezone is set', () => {
            const props = {
                ...baseProps,
                date: 1531152392,
                timeZone: 'America/New_York',
            };
            const wrapper = shallow(
                <DateHeader {...props}/>,
                {context: {intl: {formatMessage: jest.fn()}}},
            );

            expect(wrapper.getElement()).toMatchSnapshot();
        });
    });
});
