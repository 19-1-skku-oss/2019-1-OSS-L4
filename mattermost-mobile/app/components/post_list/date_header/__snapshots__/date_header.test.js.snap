// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateHeader component should match snapshot when timezone is set 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 28,
        "marginTop": 8,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#3d3c40",
        "flex": 1,
        "height": 1,
        "opacity": 0.2,
      }
    }
  />
  <View
    style={
      Object {
        "marginHorizontal": 15,
      }
    }
  >
    <InjectIntl(FormattedDate)
      day="2-digit"
      month="short"
      style={
        Object {
          "color": "#3d3c40",
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
      value={1970-01-18T17:19:12.392Z}
      weekday="short"
      year="numeric"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#3d3c40",
        "flex": 1,
        "height": 1,
        "opacity": 0.2,
      }
    }
  />
</View>
`;

exports[`DateHeader component should match snapshot with suffix 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 28,
        "marginTop": 8,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#3d3c40",
        "flex": 1,
        "height": 1,
        "opacity": 0.2,
      }
    }
  />
  <View
    style={
      Object {
        "marginHorizontal": 15,
      }
    }
  >
    <InjectIntl(FormattedDate)
      day="2-digit"
      month="short"
      style={
        Object {
          "color": "#3d3c40",
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
      value={1970-01-18T17:19:12.392Z}
      weekday="short"
      year="numeric"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#3d3c40",
        "flex": 1,
        "height": 1,
        "opacity": 0.2,
      }
    }
  />
</View>
`;

exports[`DateHeader component should match snapshot without suffix 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 28,
        "marginTop": 8,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#3d3c40",
        "flex": 1,
        "height": 1,
        "opacity": 0.2,
      }
    }
  />
  <View
    style={
      Object {
        "marginHorizontal": 15,
      }
    }
  >
    <InjectIntl(FormattedDate)
      day="2-digit"
      month="short"
      style={
        Object {
          "color": "#3d3c40",
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
      value={1970-01-18T17:19:12.392Z}
      weekday="short"
      year="numeric"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#3d3c40",
        "flex": 1,
        "height": 1,
        "opacity": 0.2,
      }
    }
  />
</View>
`;
